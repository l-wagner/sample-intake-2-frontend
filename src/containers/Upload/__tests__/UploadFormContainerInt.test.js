import React, { Component } from 'react'

import thunk from 'redux-thunk'
import multi from 'redux-multi'
import moxios from 'moxios'

import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import TestUploadFormContainer from '../UploadFormContainer'

import { Provider } from 'react-redux'
// import { uploadFormActions } from '../../../actions/'
// import uploadFormReducer from '../../../reducers/upload/uploadFormReducer'
import configureStore from '../../../store/configureStore'

import { initialFormContainerStateMock } from '../../../mocks'

// const testStore = initialFormStateMock

// const middlewares = [thunk, multi]
// const mockStore = configureStore(middlewares)

const store = configureStore
const UploadFormContainer = props => (
  <Provider store={store}>
    <TestUploadFormContainer />
  </Provider>
)

describe('Render UploadFormContainer', () => {
  it('render UploadFormContainer correctly', () => {
    const UploadFormContainerComponent = renderer
      .create(<UploadFormContainer />)
      .toJSON()
    expect(UploadFormContainerComponent).toMatchSnapshot()
  })
})

describe('responds to input', () => {
  it('has species field', () => {
    const wrapper = mount(<UploadFormContainer />)
    expect(wrapper.find('#material').exists()).toBe(true)
  })

  // it('has species field', () => {
  //   const onChangeMock = jest.fn()
  //   const wrapper = mount(<UploadFormContainer />)
  //   const materialInput = wrapper.find('#material').first()

  //   const event = { target: { value: 'Cells' } }
  //   materialInput.simulate('change', event)
  //   console.log(store.getState())
  //   console.log(materialInput.value)
  //   expect(store.getState().upload.form.selectedMaterial).toEqual('Cells')
  // })
})

// describe('should handle combinations', () =>{
// it('filters containers on material changes', () => {

// })

// })
// })
