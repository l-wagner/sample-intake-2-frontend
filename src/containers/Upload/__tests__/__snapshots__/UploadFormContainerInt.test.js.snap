// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UploadFormContainer render UploadFormContainer correctly 1`] = `
<div
  className="UploadForm-form-2"
>
  <form
    className="UploadForm-container-1"
    id="upload-form"
    onSubmit={[Function]}
  >
    <div
      className="Dropdown-textField-9"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-0-label"
        aria-owns={null}
        role="combobox"
      >
        <div
          className="MuiFormControl-root-10 MuiFormControl-fullWidth-13"
        >
          <label
            className="MuiFormLabel-root-25 MuiInputLabel-root-14 MuiInputLabel-formControl-19 MuiInputLabel-animated-22"
            data-shrink={false}
            htmlFor="downshift-0-input"
            id="downshift-0-label"
          >
            Missing translationId: upload.form.material_label for language: \${ languageCode }
          </label>
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-0-label"
            className="MuiInputBase-root-45 MuiInput-root-32 MuiInput-underline-36 MuiInputBase-formControl-46 MuiInput-formControl-33 MuiInputBase-adornedEnd-50"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              className="MuiInputBase-input-55 MuiInput-input-40 MuiInputBase-inputAdornedEnd-61"
              disabled={false}
              id="material"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              required={false}
              type="text"
              value=""
            />
            <div
              className="MuiInputAdornment-root-62 MuiInputAdornment-positionEnd-65"
            >
              <button
                aria-label="Toggle menu open"
                className="MuiButtonBase-root-73 MuiIconButton-root-67"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-72"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-76"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-209"
                />
              </button>
            </div>
          </div>
          <div
            className="MuiLinearProgress-root-216 MuiLinearProgress-colorPrimary-217 MuiLinearProgress-indeterminate-220"
            role="progressbar"
            style={
              Object {
                "bottom": 2,
                "height": 2,
                "marginBottom": -2,
                "position": "relative",
              }
            }
          >
            <div
              className="MuiLinearProgress-bar-226 MuiLinearProgress-barColorPrimary-227 MuiLinearProgress-bar1Indeterminate-229"
              style={Object {}}
            />
            <div
              className="MuiLinearProgress-bar-226 MuiLinearProgress-barColorPrimary-227 MuiLinearProgress-bar2Indeterminate-232"
              style={Object {}}
            />
          </div>
          <p
            className="MuiFormHelperText-root-85"
          >
            Missing translationId: upload.form.material_helptext for language: \${ languageCode } (0 choices)
          </p>
        </div>
      </div>
    </div>
    <div
      className="Dropdown-textField-9"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-1-label"
        aria-owns={null}
        role="combobox"
      >
        <div
          className="MuiFormControl-root-10 MuiFormControl-fullWidth-13"
        >
          <label
            className="MuiFormLabel-root-25 MuiInputLabel-root-14 MuiInputLabel-formControl-19 MuiInputLabel-animated-22"
            data-shrink={false}
            htmlFor="downshift-1-input"
            id="downshift-1-label"
          >
            Missing translationId: upload.form.application_label for language: \${ languageCode }
          </label>
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-1-label"
            className="MuiInputBase-root-45 MuiInput-root-32 MuiInput-underline-36 MuiInputBase-formControl-46 MuiInput-formControl-33 MuiInputBase-adornedEnd-50"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              className="MuiInputBase-input-55 MuiInput-input-40 MuiInputBase-inputAdornedEnd-61"
              disabled={false}
              id="application"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              required={false}
              type="text"
              value=""
            />
            <div
              className="MuiInputAdornment-root-62 MuiInputAdornment-positionEnd-65"
            >
              <button
                aria-label="Toggle menu open"
                className="MuiButtonBase-root-73 MuiIconButton-root-67"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-72"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-76"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-209"
                />
              </button>
            </div>
          </div>
          <div
            className="MuiLinearProgress-root-216 MuiLinearProgress-colorPrimary-217 MuiLinearProgress-indeterminate-220"
            role="progressbar"
            style={
              Object {
                "bottom": 2,
                "height": 2,
                "marginBottom": -2,
                "position": "relative",
              }
            }
          >
            <div
              className="MuiLinearProgress-bar-226 MuiLinearProgress-barColorPrimary-227 MuiLinearProgress-bar1Indeterminate-229"
              style={Object {}}
            />
            <div
              className="MuiLinearProgress-bar-226 MuiLinearProgress-barColorPrimary-227 MuiLinearProgress-bar2Indeterminate-232"
              style={Object {}}
            />
          </div>
          <p
            className="MuiFormHelperText-root-85"
          >
            Missing translationId: upload.form.application_helptext for language: \${ languageCode } (0 choices)
          </p>
        </div>
      </div>
    </div>
    <div
      className="Dropdown-textField-9"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-2-label"
        aria-owns={null}
        role="combobox"
      >
        <div
          className="MuiFormControl-root-10 MuiFormControl-fullWidth-13"
        >
          <label
            className="MuiFormLabel-root-25 MuiInputLabel-root-14 MuiInputLabel-formControl-19 MuiInputLabel-animated-22"
            data-shrink={false}
            htmlFor="downshift-2-input"
            id="downshift-2-label"
          >
            Missing translationId: upload.form.container_label for language: \${ languageCode }
          </label>
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-2-label"
            className="MuiInputBase-root-45 MuiInput-root-32 MuiInput-underline-36 MuiInputBase-formControl-46 MuiInput-formControl-33 MuiInputBase-adornedEnd-50"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              className="MuiInputBase-input-55 MuiInput-input-40 MuiInputBase-inputAdornedEnd-61"
              disabled={false}
              id="container"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              required={false}
              type="text"
              value=""
            />
            <div
              className="MuiInputAdornment-root-62 MuiInputAdornment-positionEnd-65"
            >
              <button
                aria-label="Toggle menu open"
                className="MuiButtonBase-root-73 MuiIconButton-root-67"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-72"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-76"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-209"
                />
              </button>
            </div>
          </div>
          <div
            className="MuiLinearProgress-root-216 MuiLinearProgress-colorPrimary-217 MuiLinearProgress-indeterminate-220"
            role="progressbar"
            style={
              Object {
                "bottom": 2,
                "height": 2,
                "marginBottom": -2,
                "position": "relative",
              }
            }
          >
            <div
              className="MuiLinearProgress-bar-226 MuiLinearProgress-barColorPrimary-227 MuiLinearProgress-bar1Indeterminate-229"
              style={Object {}}
            />
            <div
              className="MuiLinearProgress-bar-226 MuiLinearProgress-barColorPrimary-227 MuiLinearProgress-bar2Indeterminate-232"
              style={Object {}}
            />
          </div>
          <p
            className="MuiFormHelperText-root-85"
          >
            Missing translationId: upload.form.container_helptext for language: \${ languageCode }
          </p>
        </div>
      </div>
    </div>
    <div
      className="Dropdown-textField-9"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-3-label"
        aria-owns={null}
        role="combobox"
      >
        <div
          className="MuiFormControl-root-10 MuiFormControl-fullWidth-13"
        >
          <label
            className="MuiFormLabel-root-25 MuiInputLabel-root-14 MuiInputLabel-formControl-19 MuiInputLabel-animated-22"
            data-shrink={false}
            htmlFor="downshift-3-input"
            id="downshift-3-label"
          >
            Missing translationId: upload.form.species_label for language: \${ languageCode }
          </label>
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-3-label"
            className="MuiInputBase-root-45 MuiInput-root-32 MuiInput-underline-36 MuiInputBase-formControl-46 MuiInput-formControl-33 MuiInputBase-adornedEnd-50"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              className="MuiInputBase-input-55 MuiInput-input-40 MuiInputBase-inputAdornedEnd-61"
              disabled={false}
              id="species"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              required={false}
              type="text"
              value=""
            />
            <div
              className="MuiInputAdornment-root-62 MuiInputAdornment-positionEnd-65"
            >
              <button
                aria-label="Toggle menu open"
                className="MuiButtonBase-root-73 MuiIconButton-root-67"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-72"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-76"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-209"
                />
              </button>
            </div>
          </div>
          <p
            className="MuiFormHelperText-root-85"
          >
            Missing translationId: upload.form.species_helptext for language: \${ languageCode } (0 choices)
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiFormControl-root-10 Input-textField-94"
    >
      <label
        className="MuiFormLabel-root-25 MuiInputLabel-root-14 MuiInputLabel-formControl-19 MuiInputLabel-animated-22"
        data-shrink={false}
        htmlFor="number_of_samples"
      >
        Missing translationId: upload.form.number_of_samples_label for language: \${ languageCode }
      </label>
      <div
        className="MuiInputBase-root-45 MuiInput-root-32 MuiInput-underline-36 MuiInputBase-formControl-46 MuiInput-formControl-33"
        onClick={[Function]}
      >
        <input
          aria-describedby="number_of_samples-helper-text"
          aria-invalid={false}
          className="MuiInputBase-input-55 MuiInput-input-40 MuiInputBase-inputType-58 MuiInput-inputType-43"
          disabled={false}
          id="number_of_samples"
          min={0}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="number"
        />
      </div>
      <p
        className="MuiFormHelperText-root-85"
        id="number_of_samples-helper-text"
      >
        Missing translationId: upload.form.number_of_samples_helptext for language: \${ languageCode }
      </p>
    </div>
    <fieldset
      className="MuiFormControl-root-10"
    >
      <div
        className="MuiFormControl-root-10 Input-textField-94"
      >
        <label
          className="MuiFormLabel-root-25 MuiFormLabel-filled-29 MuiInputLabel-root-14 MuiInputLabel-formControl-19 MuiInputLabel-animated-22 MuiInputLabel-shrink-21"
          data-shrink={true}
          htmlFor="igo_request_id"
        >
          Missing translationId: upload.form.igo_request_id_label for language: \${ languageCode }
        </label>
        <div
          className="MuiInputBase-root-45 MuiInput-root-32 MuiInput-underline-36 MuiInputBase-formControl-46 MuiInput-formControl-33 MuiInputBase-adornedStart-49"
          onClick={[Function]}
        >
          <div
            className="MuiInputAdornment-root-62 MuiInputAdornment-positionStart-64"
          >
            <p
              className="MuiTypography-root-95 MuiTypography-body1-104 MuiTypography-colorTextSecondary-128"
            >
              IGO-
            </p>
          </div>
          <input
            aria-describedby="igo_request_id-helper-text"
            aria-invalid={false}
            className="MuiInputBase-input-55 MuiInput-input-40 MuiInputBase-inputType-58 MuiInput-inputType-43 MuiInputBase-inputAdornedStart-60"
            disabled={false}
            id="igo_request_id"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="number"
            value="444444"
          />
        </div>
        <p
          className="MuiFormHelperText-root-85 MuiFormHelperText-filled-91"
          id="igo_request_id-helper-text"
        >
          Missing translationId: upload.form.igo_request_id_helptext for language: \${ languageCode }
        </p>
      </div>
      <div
        className="CheckboxComponent-container-131"
      >
        <label
          className="MuiFormControlLabel-root-134"
        >
          <span
            className="MuiButtonBase-root-73 MuiIconButton-root-67 MuiPrivateSwitchBase-root-146 MuiCheckbox-root-140 MuiCheckbox-colorSecondary-145"
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label-72"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-76 MuiSvgIcon-fontSizeSmall-83"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
              <input
                checked={false}
                className="MuiPrivateSwitchBase-input-149"
                data-indeterminate={false}
                disabled={false}
                onChange={[Function]}
                type="checkbox"
                value="igo_alternative_id"
              />
            </span>
            <span
              className="MuiTouchRipple-root-209"
            />
          </span>
          <span
            className="MuiTypography-root-95 MuiTypography-body1-104 MuiFormControlLabel-label-139"
          >
            Missing translationId: upload.form.igo_alternative_id_label for language: \${ languageCode }
          </span>
        </label>
        <div>
          <div
            className="MuiPaper-root-151 MuiPaper-elevation1-154 MuiPaper-rounded-152 MuiCard-root-150 CheckboxComponent-card-132"
            style={
              Object {
                "opacity": 0,
              }
            }
          >
            <div
              className="MuiCardContent-root-178 CheckboxComponent-content-133"
            >
              <p
                className="MuiTypography-root-95 MuiTypography-body1-104 MuiTypography-colorTextSecondary-128"
              >
                Missing translationId: upload.form.igo_alternative_id_helptext for language: \${ languageCode }
                 
              </p>
            </div>
          </div>
        </div>
      </div>
    </fieldset>
  </form>
  <div
    className="Button-wrapper-180"
  >
    <button
      className="MuiButtonBase-root-73 MuiButton-root-183 MuiButton-contained-194 MuiButton-containedSecondary-196 MuiButton-raised-197 MuiButton-raisedSecondary-199 Button-button-179"
      disabled={false}
      form="upload-form"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="submit"
    >
      <span
        className="MuiButton-label-184"
      >
        Missing translationId: upload.form.generate_button for language: \${ languageCode }
      </span>
      <span
        className="MuiTouchRipple-root-209"
      />
    </button>
    <div
      className="Button-nothingToChange-182"
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      Missing translationId: upload.form.nothing_to_change for language: \${ languageCode }
    </div>
  </div>
   
</div>
`;
